java-checks/src/main/java/org/sonar/java/checks/helpers/IdentifierUtils.java

```diff
@@ -51,4 +56,15 @@ private static boolean isStrictAssignmentOrDeclaration(ExpressionTree expression
     }
     return true;
   }
+
+  public static class ValueAndExpression <T> {
```

Comment:
@benzonico Don't really like this name but couldn't think of a better one. Moreover, probably it would be better to have a List of ExpressionTree representing the "resolution path" but I prefer to wait to work on the ticket about refactoring of `IdentifierUtils`.